platform/darwin/src/MGLStyleLayer.h

```diff
@@ -29,7 +29,10 @@ NS_ASSUME_NONNULL_BEGIN
  
  After initializing and configuring the style layer, add it to a map view’s
  style using the `-[MGLStyle addLayer:]` or
- `-[MGLStyle insertLayer:belowLayer:]` method.
+ `-[MGLStyle insertLayer:belowLayer:]` method. Should not be used to create an 
+ instance of MGLStyleLayer directly. Instead, create instances of
+ `MGLBackgroundStyleLayer` and the concrete subclasses of
+ `MGLForegroundStyleLayer`.
```

Comment:
Actually, this is the one layer file that isn’t autogenerated, so you can put the additional text directly in here. Sorry for the confusion.